### PR TITLE
Input components (#5)

### PR DESCRIPTION
**Pull request by liquorburn**
As title says. Happy Hacktoberfest!

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -2,16 +2,26 @@
   <div id="app">
```
Comment (lubien) [inline src/App.vue]: This file should not be modified

File: src/components/PInput.vue [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<template>
+  <input v-on="$listeners"
```
Comment (lubien) [inline src/components/PInput.vue:2]: Let's indent like this:

```
  <input v-on="$listeners"
    :disabled="disabled"
    :class="classNames"
    :type="type"
    :value="value"
  >
```

File: src/components/PInput.vue [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+</template>
+
+<script>
+export default {
+  name: 'PInput',
+
+  props: {
+    type: {
+      type: String,
+      required: true
```
Comment (lubien) [inline src/components/PInput.vue]: Leave default as 'text', don't require

File: src/components/PInput.vue [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    },
+
+    value: {
+      type: String,
+      default: ''
```
Comment (lubien) [inline src/components/PInput.vue]: We shouldn't default this, require please

File: src/components/PInput.vue [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    disabled: {
+      type: Boolean,
+      default: false,
+      required: false
+    },
+
+    checked: {
+      type: Boolean,
+      default: false,
+      required: false
```
Comment (lubien) [inline src/components/PInput.vue]: No need to require if we have default

**Review by lubien — CHANGES_REQUESTED** [review]

**Comment by lubien** [comment]
Please create ´stories/PInput.js` just like the PButton version, fix the minor stuff from review and you're good to go!

File: src/App.vue (hunk 1 above)
Comment (liquorburn) [reply src/App.vue]: Done.

File: src/components/PInput.vue (hunk 2 above)
Comment (liquorburn) [reply src/components/PInput.vue:2]: Done.

File: src/components/PInput.vue (hunk 4 above)
Comment (liquorburn) [reply src/components/PInput.vue]: Done.

File: src/components/PInput.vue (hunk 5 above)
Comment (liquorburn) [reply src/components/PInput.vue]: Done.

File: src/components/PInput.vue (hunk 3 above)
Comment (liquorburn) [reply src/components/PInput.vue]: Do you mean something like this?
```
type: {
  type: String,
  default: 'text',
  required: false
},
```

**Comment by liquorburn** [comment]
> Please create ´stories/PInput.js` just like the PButton version, fix the minor stuff from review and you're good to go!

I guess the right file name is `stories/PInput.stories.js` just like `stories/PButton.stories.js`, isn't it?

File: src/App.vue [hunk 6]
```diff
@@ -7,11 +7,12 @@
 
 <script>
 import HelloWorld from './components/HelloWorld.vue'
-
+import PInput from './components/PInput.vue'
```
Comment (lubien) [inline src/App.vue]: This file should not be modified in any way :)

File: src/App.vue (hunk 1 above)
Comment (lubien) [reply src/App.vue]: You missed some parts

File: src/components/PInput.vue [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+<script>
+export default {
+  name: 'PInput',
+
+  props: {
+    type: {
+      type: String,
+      default: 'text',
+      required: false
```
Comment (lubien) [inline src/components/PInput.vue]: Remove this line

File: src/components/PInput.vue [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+    value: {
+      type: String,
+      required: true
+    },
+
+    disabled: {
+      type: Boolean,
+      default: false,
+      required: false
```
Comment (lubien) [inline src/components/PInput.vue]: Remove this line

File: stories/PInput.stories.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import PInput from '../src/components/PInput.vue';
+import Render from './Render';
+
+export default {
+  title: 'PInput',
+};
+
+export const basicUsage = () => Render({
+  components: { PInput },
+  template: '<PInput @focus="action" name="test_input" value="">',
```
Comment (lubien) [inline stories/PInput.stories.js]: You need to create a prop named `name` on `src/components/PInput.vue` too.

Comment (lubien) [inline stories/PInput.stories.js]: `value` prop should actually be v-model. [Read more about it](https://vuejs.org/v2/guide/components.html#Using-v-model-on-Components) then please update **all** examples.

The please put an `@input` just to log changes on v-model

File: stories/PInput.stories.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+export const inputCheckboxDisabled = () => Render({
+  components: { PInput },
+  template: '<PInput type="checkbox" name="test_input" value="" disabled>',
+  methods: { }
+});
+
+export const inputRadio = () => Render({
+  components: { PInput },
+  template: '<PInput @click="action" type="radio" name="test_input" value="">',
```
Comment (lubien) [inline stories/PInput.stories.js]: Weird to have a click event on an input. Use `@input` again please like I mentioned above

**Review by lubien — CHANGES_REQUESTED** [review]

**Comment by lubien** [comment]
> I guess the right file name is `stories/PInput.stories.js` just like `stories/PButton.stories.js`, isn't it?

Yes you're right :)

File: src/App.vue [hunk 11]
```diff
@@ -7,7 +7,6 @@
 
 <script>
 import HelloWorld from './components/HelloWorld.vue'
-
```
Comment (lubien) [inline src/App.vue]: Let's keep this file unchanged :(

File: src/components/PInput.vue (hunk 3 above)
Comment (lubien) [reply src/components/PInput.vue]:
```
type: {
  type: String,
  default: 'text'
},
```

Please don't mark as solved if it's not

File: stories/PInput.stories.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+import PInput from '../src/components/PInput.vue';
+import Render from './Render';
+
+export default {
+  title: 'PInput'
+};
+
+export const basicUsage = () => Render({
+  components: { PInput },
+  template: '<PInput @focus="action" name="input_basic" v-model="value" @input="myLog" />',
```
Comment (lubien) [inline stories/PInput.stories.js]: Don't forget to create a `data` and have `value` there as an empty string on all examples.

Would be great if `myLog` logged `value`.

**Review by lubien — CHANGES_REQUESTED** [review]

File: stories/PInput.stories.js (hunk 12 above)
Comment (liquorburn) [reply stories/PInput.stories.js]: I really can't get `value` logged... Any help?

File: src/App.vue (hunk 11 above)
Comment (liquorburn) [reply src/App.vue]: You're right sorry... Guess now it's ok.

File: src/components/PInput.vue (hunk 3 above)
Comment (liquorburn) [reply src/components/PInput.vue]: Done.

File: stories/PInput.stories.js (hunk 12 above)
Comment (lubien) [reply stories/PInput.stories.js]: Something like this:

```
export const basicUsage = () => Render({
  components: { PInput },
  template: '<PInput @focus="action" name="input_basic" v-model="name" @input="myLog(name)" />',
  data: () =>({ name: '' }),
  methods: { action: action('focused'), myLog: action('input') }
});
```

File: src/components/PInput.vue [hunk 13]
```diff
@@ -0,0 +1,55 @@
+<template>
+  <input v-on="$listeners"
+    :disabled="disabled"
+    :class="classNames"
+    :type="type"
+    v-bind:value="value"
```
Comment (lubien) [inline src/components/PInput.vue]: You need to make a model property to make this work properly. Add:

```js
  data () {
    return {
      model: this.value
    }
  },
```

Then here change this line to

```js
v-model="model"
@input="$emit('input', $event.target.value)"
```

Also there's an issue that'd make two input listeners so update `$listeners` to `listeners`. Add:

```js
computed: {
    classNames () {
      return [
        'puto-input',
        (this.disabled && '-disabled'),
        (this.loading && '-loading')
      ]
    },

    listeners() {
      // Delete duplicate listeners
      const listeners = {...this.$listeners}
      delete listeners.input
      return listeners
    }
  }

```

**Review by lubien — CHANGES_REQUESTED** [review]

**Comment by lubien** [comment]
image

Can you add **another** example that show this works on a v-model and multiple radio buttons (meaning if I check one, others are unchecked)

**Comment by liquorburn** [comment]
> image
> 
> Can you add **another** example that show this works on a v-model and multiple radio buttons (meaning if I check one, others are unchecked)

Yes, I've just added it.

**Comment by lubien** [comment]
Thanks @liquorburn

**Pull request merged**